src/main/scala/kanaloa/reactive/dispatcher/Backend.scala

```diff
@@ -1,41 +1,62 @@
 package kanaloa.reactive.dispatcher
 
 import akka.actor.{ActorRefFactory, Actor, Props, ActorRef}
-import kanaloa.reactive.dispatcher.Backend.BackendAdaptor
 import scala.reflect._
-
 import scala.concurrent.Future
+import scala.language.implicitConversions
 
 trait Backend {
   def apply(f: ActorRefFactory): ActorRef
 }
 
-object Backend extends BackendAdaptors {
-
-  type BackendAdaptor[T] = T ⇒ Backend
+object Backend {
 
   def apply(f: ActorRefFactory ⇒ ActorRef): Backend = new Backend {
     def apply(factory: ActorRefFactory): ActorRef = f(factory)
   }
-}
 
-trait BackendAdaptors {
-  implicit val artb: BackendAdaptor[ActorRef] = ar ⇒ Backend((_: ActorRefFactory) ⇒ ar)
-  implicit val aptb: BackendAdaptor[Props] = props ⇒ Backend((f: ActorRefFactory) ⇒ f.actorOf(props))
-  implicit def aftb[ReqT: ClassTag, ResponseT]: BackendAdaptor[ReqT ⇒ Future[ResponseT]] =
-    (f: ReqT ⇒ Future[ResponseT]) ⇒ Backend((factory: ActorRefFactory) ⇒ factory.actorOf(Props(new SimpleDelegatee[ReqT, ResponseT](f))))
-
-  case class UnexpectedRequest(request: Any) extends Exception
-  private class SimpleDelegatee[ReqT: ClassTag, ResponseT](f: ReqT ⇒ Future[ResponseT]) extends Actor {
-    import context.dispatcher
-    def receive: Receive = {
-      case t: ReqT if classTag[ReqT].runtimeClass.isInstance(t) ⇒
-        val replyTo = sender
-        f(t).recover {
-          case e: Throwable ⇒ e
-        }.foreach(replyTo ! _)
-
-      case m ⇒ sender ! UnexpectedRequest(m)
+  trait BackendAdaptor[T] {
+    def apply(t: T): Backend
+  }
+
+  object BackendAdaptor extends BackendAdaptors
+
+  trait BackendAdaptors {
+
+    // helper to create an adapator from a function
+    def apply[T](f: T ⇒ Backend) = new BackendAdaptor[T] {
+      def apply(t: T): Backend = f(t)
     }
+
+    implicit val backendBackend = apply[Backend](identity)
+
+    // accepting subtypes of ActorRef to also support TestActorRef
+    implicit def actorRefBackend[T <: ActorRef] = apply[T](ref ⇒ Backend(_ ⇒ ref))
+
+    implicit val propsBackend = apply[Props](props ⇒ Backend(_.actorOf(props)))
+
+    implicit def delegateeeBackend[ReqT: ClassTag, ResponseT] = apply[ReqT ⇒ Future[ResponseT]](
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [You should not name methods after their defining object](https://www.codacy.com/app/kailuo-wang/kanaloa/file/1795474934/issues/source?bid=3018831&fileBranchId=3071693#l38)